### PR TITLE
Change button background color enhancement

### PR DESCRIPTION
**Pull request by Lentumunai-Mark**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes https://github.com/opensrp/fhircore/issues/2252

**Engineer Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/shared/components/ActionableButton.kt [hunk 1]
```diff
@@ -178,6 +183,14 @@ fun ButtonProperties.interpolateStatus(computedValuesMap: Map<String, Any>): Ser
   else ServiceStatus.UPCOMING
 }
 
+@Composable
+fun ButtonProperties.interpolateBackgroundColor(computedValuesMap: Map<String, Any>): Color {
+  val interpolated = this.backgroundColor?.interpolate(computedValuesMap)
+  return if (!interpolated.isNullOrEmpty()) {
+    return interpolated.parseColor()
+  } else Color.Unspecified
+}
```
Comment (pld) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/shared/components/ActionableButton.kt:192]: what if combine interpolation for any (or all the ones we want to) ButtonProperties into a single step, any thoughts on how we can do that?

relatedly, wondering if these interpolation functions (or a more general one) should live in ButtonProperties and not here.

relatedly, a little uneasy that we're combining interpolation and setting a default (with the else) into single step, that's a bit broader than only interpolation b/c the name is mislead -- I see this pattern was introduced before so this is not a critique of this specific addition, it's following that pattern

Comment (Lentumunai-Mark) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/shared/components/ActionableButton.kt:192]: Yeah sure, I'll move the interpolations to one function in the button properties responsible for handling all the interpolations.

**Review by pld — CHANGES_REQUESTED** [review]
lgtm!

**Review by pld — APPROVED** [review]

**Review by roywanyaga — APPROVED** [review]
looks good to me

**Pull request merged**